gameplay/src/Ref.h

```diff
@@ -43,6 +43,41 @@ class Ref
      */
     unsigned int getRefCount() const;
 
+    /**
```

Comment:
However, the entire framework is not implemented this way. Do you plan to go through thousands of lines and protect against this everywhere ? Or only your new implemented code that is using it. We need consistency in the codebase.
My point is that there are ways to avoid it by simply using is properly as-is.
The strategy you have implemented is VERY similar to what I might have implemented for a smart pointer.